### DISCUSSION TITLE
Welcome to react-telegram-web-app Discussions!

### DISCUSSION THREAD

**Discussion by vkruglikov** [vkruglikov/react-telegram-web-app#28] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by rexydye** [comment]
Hello, how can i get initdata string?

**Reply by vkruglikov** [reply]
@rexydye  This package doesn't provide API for initData.
You need to use it directly from the window.Telegram.WebApp

But that may change in the future.

**Reply by vkruglikov** [reply]
Hi! hook `useInitData` was added at the last release :)

**Comment by timseriakov** [comment]
Hi, what means _All Telegram WebApp feature support._?

**Reply by vkruglikov** [reply]
Hi! At present, we don't have some functions like `showAlert`, `showConfirm`, `requestWriteAccess` and etc. If you want to use it, you have to use it directly from `window.Telegram`

**Comment by Ives-code** [comment]
Can it be packaged separately as a web page?

**Reply by vkruglikov** [reply]
Hi! What do you mean?

**Comment by shivamgupta88** [comment]
Help Needed with React Mini-App: Blank Web Page Issue

Hello everyone,

I'm currently working on a mini-app using React and have encountered an issue where my web page displays nothing (it's completely blank). I have been following the installation and setup instructions provided in the README file of the [react-telegram-web-app](https://github.com/vkruglikov/react-telegram-web-app?tab=readme-ov-file#-installation--get-started) repository on GitHub. Despite following all the mentioned steps, my web page remains empty.

Here are two screenshots of what I'm seeing:
<img width="1680" alt="Screenshot 2024-02-13 at 16 48 08" src="https://github.com/vkruglikov/react-telegram-web-app/assets/72003102/029b87d9-ba74-4518-a4f5-31b296ee7aed">


<img width="1680" alt="Screenshot 2024-02-13 at 16 46 18" src="https://github.com/vkruglikov/react-telegram-web-app/assets/72003102/bcf9d2cf-63db-4e59-a732-e36bfdb6f1eb">

my git repo - https://github.com/shivamgupta88/mini-app-with-reactjs


I would greatly appreciate any insights, suggestions, or guidance on how to resolve this issue. If additional information or code snippets are needed, I'm happy to provide them.

Thank you in advance for your help!

**Comment by muhammad-tayyab3** [comment]
Title: How to Redirect Back to Telegram Mini App After Video Upload via Browser?

Question:
I'm working on a Telegram mini app built with ReactJS. In the app, users upload videos, but in some cases, the video doesn't load or errors occur. When an error happens, I redirect the user to upload the video through the browser. Once the video is uploaded successfully, I need the user to automatically navigate back to the same Telegram mini app tab or screen without manually opening the mini app again.

I tried using window.Telegram.WebApp.Back() and other Telegram Web App methods, but it doesn't seem to work when the user is redirected from the Telegram mini app to the browser and then back again after the upload.

Has anyone encountered this issue? What is the recommended solution for redirecting back to the Telegram mini app after a successful video upload in a browser?

Thanks in advance for any help or insights!
